src/lib/llm-provider.ts

```diff
@@ -1,12 +1,41 @@
-import { createOpenAI, openai } from "@ai-sdk/openai";
+import { openai } from "@ai-sdk/openai";
+import { createOllama, OllamaProvider } from "ollama-ai-provider";
+
+enum LLMProviderType {
+  OPENAI = "openai",
+  OLLAMA = "ollama",
+}
+
+type LLMConfiguration = {
+  provider: typeof openai | OllamaProvider;
+  model: string;
+  settings?: Record<string, unknown>;
+};
+
+const llmConfigurations: Record<LLMProviderType, LLMConfiguration> = {
+  [LLMProviderType.OPENAI]: {
+    provider: openai,
+    model: process.env.LLM_MODEL || "gpt-4o",
+  },
+  [LLMProviderType.OLLAMA]: {
+    provider: createOllama({ baseURL: process.env.OLLAMA_URL }),
```

Comment:
Please add a note to `README.md` and the `.env.defaults` on how to properly configure OLLAMA - including some default `OLLAMA_URL`

Reply:
@pkarw Great point! Added the instructions for Ollama in `README.md` and adjusted `.env.defaults` as well. 👍